src/register/register.component.ts

```diff
@@ -1,46 +1,40 @@
-import {Component, OnInit} from '@angular/core';
+import {Component} from '@angular/core';
 import {AlertService} from '../_services/alert.service';
 import {UserService} from '../_services/user.service';
-import {User} from '../_models';
+import {User} from '../_models/User';
 import {sha512} from 'js-sha512';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
 
 @Component({
   selector: 'app-register',
   templateUrl: './register.component.html',
   styleUrls: ['./register.component.css']
 })
-export class RegisterComponent implements OnInit {
-  model: User = {};
-
+export class RegisterComponent {
+  form: FormGroup = new FormGroup({
+    username: new FormControl(
+      null, [Validators.required]),
+    password: new FormControl(
+      null, [Validators.required])
+  });
   loading = false;
 
-  constructor(
-    // private router: Router,
-              private userService: UserService,
+  constructor(private userService: UserService,
               private alertService: AlertService) {
   }
 
-  register() {
+  onSubmit({username, password}) {
     this.loading = true;
-    const user: User = {username: this.model.username, password: sha512(this.model.password)};
+    const user: User = {username, password: sha512(password)};
     this.userService.create(user)
       .subscribe(
         data => {
-          this.alertService.success('Registration successful');
-          // this.router.navigate(['/login']);
-        },
-        error => {
-          this.loading = false;
-          console.log(error);
-          if (error.status === 400) {
-            this.alertService.error(error.error);
+          if (data === null) {
+            this.alertService.error('Registration unsuccessful');
```

Comment:
Let's show server error always if it exists. 
If no error message show status code description (if available somewhere).